app/controllers/posts_controller.rb

```diff
@@ -1,20 +1,22 @@
 class PostsController < ApplicationController
-  def new
-    @post = Post.new
-  end
-
-  def create
-    @post = Post.create(post_params)
-    redirect_to posts_url
-  end
 
   def index
-    @posts = Post.all
+    # this method is going to need parameters passed to it
+    @new_post = Post.new
+    @user_id = session[:user_id]
```

Comment:
I'd say this isn't needed for the sake of skinny controllers and readability, you can just refactor your view line for to call `current_user.id` (will comment there as well)

Reply:
done